src/main/java/com/iota/iri/Iota.java

```diff
@@ -40,90 +39,48 @@
-    public final Configuration configuration;
-    public final Hash coordinator;
+    public final IotaConfig configuration;
     public final TipsViewModel tipsViewModel;
     public final MessageQ messageQ;
     public final TipSelector tipsSelector;
 
-    public final boolean testnet;
-    public final int maxPeers;
-    public final int udpPort;
-    public final int tcpPort;
-    public final int maxTipSearchDepth;
-
-    public Iota(Configuration configuration) throws IOException {
+    public Iota(IotaConfig configuration) throws IOException {
         this.configuration = configuration;
-        testnet = configuration.booling(Configuration.DefaultConfSettings.TESTNET);
-        maxPeers = configuration.integer(Configuration.DefaultConfSettings.MAX_PEERS);
-        udpPort = configuration.integer(Configuration.DefaultConfSettings.UDP_RECEIVER_PORT);
-        tcpPort = configuration.integer(Configuration.DefaultConfSettings.TCP_RECEIVER_PORT);
-
-        String snapshotFile = configuration.string(Configuration.DefaultConfSettings.SNAPSHOT_FILE);
-        String snapshotSigFile = configuration.string(Configuration.DefaultConfSettings.SNAPSHOT_SIGNATURE_FILE);
-        Snapshot initialSnapshot = Snapshot.init(snapshotFile, snapshotSigFile, testnet).clone();
-        long snapshotTimestamp = configuration.longNum(Configuration.DefaultConfSettings.SNAPSHOT_TIME);
-        int milestoneStartIndex = configuration.integer(Configuration.DefaultConfSettings.MILESTONE_START_INDEX);
-        int numKeysMilestone = configuration.integer(Configuration.DefaultConfSettings.NUMBER_OF_KEYS_IN_A_MILESTONE);
-        double alpha = configuration.doubling(Configuration.DefaultConfSettings.TIPSELECTION_ALPHA.name());
-        int belowMaxDepthTxLimit = configuration.integer(
-                Configuration.DefaultConfSettings.BELOW_MAX_DEPTH_TRANSACTION_LIMIT);
-
-        boolean dontValidateMilestoneSig = configuration.booling(Configuration.DefaultConfSettings
-                .DONT_VALIDATE_TESTNET_MILESTONE_SIG);
-        int transactionPacketSize = configuration.integer(Configuration.DefaultConfSettings.TRANSACTION_PACKET_SIZE);
-
-        maxTipSearchDepth = configuration.integer(Configuration.DefaultConfSettings.MAX_DEPTH);
-        if(testnet) {
-            String coordinatorTrytes = configuration.string(Configuration.DefaultConfSettings.COORDINATOR);
-            if(StringUtils.isNotEmpty(coordinatorTrytes)) {
-                coordinator = new Hash(coordinatorTrytes);
-            } else {
-                log.warn("No coordinator address given for testnet. Defaulting to "
-                        + Configuration.TESTNET_COORDINATOR_ADDRESS);
-                coordinator = new Hash(Configuration.TESTNET_COORDINATOR_ADDRESS);
-            }
-        } else {
-            coordinator = new Hash(Configuration.MAINNET_COORDINATOR_ADDRESS);
-        }
+        Snapshot initialSnapshot = Snapshot.init(configuration.getSnapshotFile(),
```

Comment:
Going off this `init`, `Snapshot::init` is the only method that does not take an `IotaConfig` object as parameter.
Could you change this to be in line with the other calls below?
`Iota` should not need to know which config values `Snapshot` needs

Reply:
:+1: